### PR TITLE
Add javadoc link to WPILib docs and oracle java docs.

### PR DESCRIPTION
**Pull request by thenetworkgrinch**
Adds javadocs external links for javadocs generation. It may be worthwhile to add `javadocs.destinationDir = new File("$buildDir/repos/")` that way javadocs is also exported to the maven repo for an easy hosting solution.

### CODE REVIEW

**Comment by calcmogul** [comment]
Is this not already done in https://github.com/wpilibsuite/allwpilib/blob/main/docs/build.gradle#L189 ?

File: build.gradle [hunk 1]
```diff
@@ -37,6 +37,13 @@ dependencies {
     implementation 'edu.wpi.first.thirdparty.frc2024.opencv:opencv-java:4.8.0-4'
 }
 
+// Setup Javadocs to link back to WPILib docs
+javadoc {
+    options {
+        links 'https://docs.oracle.com/en/java/javase/11/docs/api/', 'https://github.wpilib.org/allwpilib/docs/release/java/'
```
Comment (rzblue) [inline build.gradle]: We use java 17

**Review by rzblue — CHANGES_REQUESTED** [review]

**Comment by sciencewhiz** [comment]
Does this break during beta if the vendor is using a new API that isn't in the WPILib release docs?

**Comment by thenetworkgrinch** [comment]
I have no clue, i was able to compile it just fine when i tested it. Javadocs usually doesnt link to anything that it cant find, so i imagine it would just do the class name with no external link like it normally does without it.

**Comment by thenetworkgrinch** [comment]
To answer your question, it appears if something doesnt exist it just wont link to it. See here as an example.

https://pathplanner.dev/api/java/com/pathplanner/lib/util/DriveFeedforwards.html#%3Cinit%3E(edu.wpi.first.units.measure.LinearAcceleration[],edu.wpi.first.units.measure.Force[],edu.wpi.first.units.measure.Current[],edu.wpi.first.units.measure.Force[],edu.wpi.first.units.measure.Force[])

File: build.gradle [hunk 2]
```diff
@@ -37,6 +37,13 @@ dependencies {
     implementation 'edu.wpi.first.thirdparty.frc2024.opencv:opencv-java:4.8.0-4'
 }
 
+// Setup Javadocs to link back to WPILib docs
+javadoc {
+    options {
+        links 'https://docs.oracle.com/en/java/javase/17/docs/api/', 'https://github.wpilib.org/allwpilib/docs/release/java/'
```
Comment (ThadHouse) [inline build.gradle]: Abstract out the version here and put it near where we set the compile version.

Comment (thenetworkgrinch) [reply build.gradle]: At the top of this script or after the wpilib version change?

Comment (ThadHouse) [reply build.gradle]: https://github.com/wpilibsuite/vendor-template/blob/2025/build.gradle#L20

Here, so we know we need to change the docs version if we change the java target.

**Review by ThadHouse — APPROVED** [review]

**Comment by thenetworkgrinch** [comment]
@ThadHouse Just fixed that compilation error

**Pull request merged**